src/App.jsx

```diff
@@ -14,33 +16,93 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FILD_ALPHABETICCALY = 'alphabeticcaly';
+const SORT_FILD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, sortField, rever) {
+  let newVisibleGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField === SORT_FILD_ALPHABETICCALY) {
+    newVisibleGoods.sort();
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortField === SORT_FILD_LENGTH) {
+    newVisibleGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (rever) {
+    return newVisibleGoods.reverse();
+  }
+
+  if (sortField === '') {
+    newVisibleGoods = [...goods];
+  }
+
+  return newVisibleGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [rever, setRever] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, rever);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortField !== SORT_FILD_ALPHABETICCALY,
+          })}
+          onClick={() => setSortField(SORT_FILD_ALPHABETICCALY)}
+        >
+          Sort alphabetically
+        </button>
```

Comment:
The checklist requires highlighting the active button. Ensure that the button's active state is visually distinguishable when the sortField is SORT_FILD_ALPHABETICCALY.